Gateway/Request/ReturnUrlDataBuilder.php

```diff
@@ -87,8 +87,7 @@ public function build(array $buildSubject)
         if ($pwaOrigin) {
             $returnUrl = rtrim($pwaOrigin, '/') . '/adyen/process/result?merchantReference=' . $order->getIncrementId();
         } else {
-            $this->url->setQueryParam('merchantReference', $order->getIncrementId());
-            $returnUrl = $this->url->getUrl("adyen/process/result");
+            $returnUrl = $this->url->getUrl("adyen/process/result", ['merchantReference' => $order->getIncrementId()]);
```

Comment:
I see that you also tested LPMs in the regular checkout, so it is confirmed that the redirect methods also work with this modification?

Reply:
Yes, I tested this modification also works in regular checkout